packages/components/Tabs/src/TabsItem.tsx

```diff
@@ -39,24 +38,32 @@ export const TabsItem = compose<TabsItemType>({
     // Grabs the context information from Tabs (currently selected TabsItem and client's onTabsClick callback).
     const info = React.useContext(TabsContext);
 
-    /* There's a bug where the user callback is being called multiple times on one click.
-    We check that there is an actual change in selection before forwarding the message to the user callback
-    so that they aren't notified more than once for each click. */
-    const changeSelection = () => {
-      if (itemKey != info.selectedKey) {
+    const [focusState, setFocusState] = React.useState({
+      focused: false,
+    });
+
+    const changeSelection = React.useCallback(() => {
+      componentRef?.current?.focus();
+    }, [componentRef]);
+
+    const changeSelectionWithFocus = React.useCallback(() => {
+      setFocusState({ focused: true });
+      if (!focusState.focused) {
         info.onTabsClick && info.onTabsClick(itemKey);
         info.getTabId && info.getTabId(itemKey, info.tabsItemKeys.findIndex(x => x == itemKey) + 1);
         info.updateSelectedTabsItemRef && componentRef && info.updateSelectedTabsItemRef(componentRef);
       }
-    };
+    }, [focusState, setFocusState, componentRef, info]);
```

Comment:
Actually, this would also need `itemKey`. Sorry I missed that. Every var/function within a useCallback should be in the dependency array, because if one of them changes, the output of the function changes.